k2/python/k2/fsa.py

```diff
@@ -70,23 +117,78 @@ def __setattr__(self, name: str, value: Any) -> None:
             object.__setattr__(self, name, value)
         elif isinstance(value, torch.Tensor):
             assert value.shape[0] == self.arcs.values().shape[0]
-            # TODO(fangjun): do we really need a copy here?
+            if name == 'labels':
+                assert value.dtype == torch.int32
+                self.arcs.values()[:, 2] = value
+                return
+
             self._tensor_attr[name] = value.clone()
+
             if name == 'score':
                 assert value.dtype == torch.float32
+                # NOTE: we **reinterpret** the float patterns
+                # to integer patterns here.
                 self.arcs.values()[:, -1] = _as_int(value.detach())
         else:
-            # TODO(fangjun): do we need to copy value?
             self._non_tensor_attr[name] = value
 
     def __getattr__(self, name: str) -> Any:
-        if name in self._tensor_attr:
+        if name == 'labels':
+            return self.arcs.values()[:, 2]
+        elif name in self._tensor_attr:
             return self._tensor_attr[name]
         elif name in self._non_tensor_attr:
             return self._non_tensor_attr[name]
 
         raise AttributeError(f'Unknown attribute {name}')
 
+    def __delattr__(self, name: str) -> None:
+        assert name not in ('arcs', 'score', 'labels')
```

Comment:
I think 'score' should be 'scores', since all of the other attributes are plurals.

Reply:
Fixed.